main/params.go

```diff
@@ -321,75 +332,117 @@ func init() {
+		Config.StakingKeyFile = stakingKeyPath
+	}
+	stakingCertPath := v.GetString(stakingCertPathKey)
+	if stakingCertPath == defaultString {
+		Config.StakingCertFile = defaultStakingCertPath
+	} else {
+		Config.StakingCertFile = stakingCertPath
+	}
+
+	// parse any env variables
+	Config.StakingCertFile = os.ExpandEnv(Config.StakingCertFile)
+	Config.StakingKeyFile = os.ExpandEnv(Config.StakingKeyFile)
+	switch {
+	// If staking key/cert locations are specified but not found, error
+	case Config.StakingKeyFile != defaultStakingKeyPath || Config.StakingCertFile != defaultStakingCertPath:
+		if _, err := os.Stat(Config.StakingKeyFile); os.IsNotExist(err) {
+			return fmt.Errorf("couldn't find staking key at %s", Config.StakingKeyFile)
+		} else if _, err := os.Stat(Config.StakingCertFile); os.IsNotExist(err) {
+			return fmt.Errorf("couldn't find staking certificate at %s", Config.StakingCertFile)
+		}
+	default:
+		// Only creates staking key/cert if [stakingKeyPath] doesn't exist
+		if err := staking.GenerateStakingKeyCert(Config.StakingKeyFile, Config.StakingCertFile); err != nil {
+			return fmt.Errorf("couldn't generate staking key/cert: %w", err)
+		}
+	}
 
 	// Bootstrapping:
-	if *bootstrapIPs == "default" {
-		*bootstrapIPs = strings.Join(defaultBootstrapIPs, ",")
+	defaultBootstrapIPs, defaultBootstrapIDs := genesis.SampleBeacons(Config.NetworkID, 5)
+	bootstrapIPs := v.GetString(bootstrapIPsKey)
+	if bootstrapIPs == defaultString {
+		bootstrapIPs = strings.Join(defaultBootstrapIPs, ",")
 	}
-	for _, ip := range strings.Split(*bootstrapIPs, ",") {
+	for _, ip := range strings.Split(bootstrapIPs, ",") {
 		if ip != "" {
 			addr, err := utils.ToIPDesc(ip)
 			if err != nil {
-				errs.Add(fmt.Errorf("couldn't parse ip: %w", err))
-				return
+				return fmt.Errorf("couldn't parse ip: %w", err)
 			}
 			Config.BootstrapPeers = append(Config.BootstrapPeers, &node.Peer{
 				IP: addr,
 			})
 		}
 	}
 
-	if *bootstrapIDs == "default" {
-		if *bootstrapIPs == "" {
-			*bootstrapIDs = ""
+	bootstrapIDs := v.GetString(bootstrapIDsKey)
+	if bootstrapIDs == defaultString {
+		if bootstrapIPs == "" {
+			bootstrapIDs = ""
 		} else {
-			*bootstrapIDs = strings.Join(defaultBootstrapIDs, ",")
+			bootstrapIDs = strings.Join(defaultBootstrapIDs, ",")
 		}
 	}
 
 	if Config.EnableStaking && !Config.EnableP2PTLS {
-		errs.Add(errStakingRequiresTLS)
-		return
+		return errStakingRequiresTLS
 	}
 
 	if !Config.EnableStaking && Config.DisabledStakingWeight == 0 {
-		errs.Add(errInvalidStakerWeights)
+		return errInvalidStakerWeights
 	}
 
 	if Config.EnableP2PTLS {
 		i := 0
-		for _, id := range strings.Split(*bootstrapIDs, ",") {
+		for _, id := range strings.Split(bootstrapIDs, ",") {
 			if id != "" {
 				peerID, err := ids.ShortFromPrefixedString(id, constants.NodeIDPrefix)
 				if err != nil {
-					errs.Add(fmt.Errorf("couldn't parse bootstrap peer id: %w", err))
-					return
+					return fmt.Errorf("couldn't parse bootstrap peer id: %w", err)
```

Comment:
Let's we print the ID we can't parse:

`fmt.Errorf("couldn't parse bootstrap peer id '%s': %w", id, err)`

Reply:
updated